components/arm/xarm/xarm.go

```diff
@@ -23,26 +22,40 @@ import (
 	"go.viam.com/rdk/utils"
 )
 
-// AttrConfig is used for converting config attributes.
-type AttrConfig struct {
+// Config is used for converting config attributes.
+type Config struct {
 	Host         string  `json:"host"`
+	Port         int     `json:"port"`
 	Speed        float32 `json:"speed_degs_per_sec"`
 	Acceleration float32 `json:"acceleration_degs_per_sec_per_sec"`
+
+	parsedPort string
+}
+
+// Validate validates the config.
+func (cfg *Config) Validate(path string) ([]string, error) {
+	var deps []string
+	if cfg.Port == 0 {
+		cfg.parsedPort = defaultPort
+	} else {
+		cfg.parsedPort = fmt.Sprintf("%d", cfg.Port)
+	}
+	return deps, nil
 }
 
 const (
 	defaultSpeed        = 20
 	defaultAcceleration = 50
-	defaultPort         = ":502"
+	defaultPort         = "502"
 )
 
 type xArm struct {
-	generic.Unimplemented
+	resource.Named
+	mu       sync.RWMutex
```

Comment:
How can I tell what members are protected by this mutex?

Reply:
its anything set in reconfigure

Reply:
How would someone continuing to write code for this type know that?

Reply:
Since its only around configured things, modifying a configuration will require one to read Reconfigure and see the lock and its usage implying that the variables changes are mutually exclusive in a read/write lock. Further, running tests with -race will catch issues of code that is not protected which has caught numerous issues.

Reply:
Also I don't see anywhere that we are taking the mutex to access conn and speed.